stdlib/public/core/Codable.swift

```diff
@@ -0,0 +1,3176 @@
+
+    /// Encodes the given value for the given key.
+    ///
+    /// - parameter value: The value to encode.
+    /// - parameter key: The key to associate the value with.
+    /// - throws: `EncodingError.invalidValue` if the given value is invalid in the current context for this format.
+    mutating func encode(_ value: String?, forKey key: Key) throws
+
+    /// Encodes the given object weakly for the given key.
+    ///
+    /// For `Encoder`s that implement this functionality, this will only encode the given object and associate it with the given key if it is encoded unconditionally elsewhere in the payload (either previously or in the future).
+    ///
+    /// For formats which don't support this feature, the default implementation encodes the given object unconditionally.
+    ///
+    /// - parameter object: The object to encode.
+    /// - parameter key: The key to associate the object with.
+    /// - throws: `EncodingError.invalidValue` if the given value is invalid in the current context for this format.
+    mutating func encodeWeak<T : AnyObject & Encodable>(_ object: T?, forKey key: Key) throws
+
+    /// Stores a keyed encoding container for the given key and returns it.
+    ///
+    /// - parameter keyType: The key type to use for the container.
+    /// - parameter key: The key to encode the container for.
+    /// - returns: A new keyed encoding container.
+    mutating func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type, forKey key: Key) -> KeyedEncodingContainer<NestedKey>
+
+    /// Stores an unkeyed encoding container for the given key and returns it.
+    ///
+    /// - parameter key: The key to encode the container for.
+    /// - returns: A new unkeyed encoding container.
+    mutating func nestedUnkeyedContainer(forKey key: Key) -> UnkeyedEncodingContainer
+
+    /// Stores a new nested container for the default `super` key and returns a new `Encoder` instance for encoding `super` into that container.
+    ///
+    /// Equivalent to calling `superEncoder(forKey:)` with `Key(stringValue: "super", intValue: 0)`.
+    ///
+    /// - returns: A new `Encoder` to pass to `super.encode(to:)`.
+    mutating func superEncoder() -> Encoder
+
+    /// Stores a new nested container for the given key and returns a new `Encoder` instance for encoding `super` into that container.
+    ///
+    /// - parameter key: The key to encode `super` for.
+    /// - returns: A new `Encoder` to pass to `super.encode(to:)`.
+    mutating func superEncoder(forKey key: Key) -> Encoder
+}
+
+// An implementation of _KeyedEncodingContainerBase and _KeyedEncodingContainerBox are given at the bottom of this file.
+/// `KeyedEncodingContainer` is a type-erased box for `KeyedEncodingContainerProtocol` types, similar to `AnyCollection` and `AnyHashable`. This is the type which consumers of the API interact with directly.
+public struct KeyedEncodingContainer<K : CodingKey> {
```

Comment:
It can, but I don't see much of a benefit to doing so. When would you want to use it as `KeyedEncodingContainerProtocol` instead of the fully-typed `KeyedEncodingContainer<K>`?